frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/traverse/NextBranchCheckoutAction.java

```diff
@@ -0,0 +1,38 @@
+package com.virtuslab.gitmachete.frontend.actions.traverse;
+
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import lombok.experimental.ExtensionMethod;
+import lombok.val;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+import com.virtuslab.gitmachete.frontend.actions.base.BaseCheckoutAction;
+import com.virtuslab.gitmachete.frontend.actions.expectedkeys.IExpectsKeySelectedBranchName;
+import com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle;
+
+@ExtensionMethod(GitMacheteBundle.class)
+public class NextBranchCheckoutAction extends BaseCheckoutAction
+    implements
+      IExpectsKeySelectedBranchName {
+
+  @Override
+  protected @Nullable String getTargetBranchName(AnActionEvent anActionEvent) {
+    val gitRepository = getSelectedGitRepository(anActionEvent);
+    val currentBranchName = getCurrentBranchNameIfManaged(anActionEvent);
+    if (gitRepository != null) {
+      val localBranches = gitRepository.getBranches().getLocalBranches().iterator();
+      while (localBranches.hasNext()) {
+        val nextBranch = localBranches.next();
```

Comment:
Whoops it should check out the next branch in the list of branches returned by `gitRepository.getBranches().getLocalBranches()`... but what we aim for is to check out the next branch in the **branch layout** as managed by git-machete!

Reply:
Actually! that's a good moment to cover these new actions with some UI tests. I'd replace some calls to `project.checkoutBranch(...)` with `project.checkout{Next,Previous,Parent}Branch()` which would call your newly-added actions. `project.checkout...Branch()` method would need to be added in both Scala and Rhino JS side.

Before the work on UI tests, pls make sure you've got all the useful IntelliJ plugins enabled/installed, as per CONTRIBUTING.md